### PR TITLE
Adapt to the Python 3.12 final release

### PR DESCRIPTION
**Pull request by wRAR**

### CODE REVIEW

File: .github/workflows/tests-windows.yml [hunk 1]
```diff
@@ -21,13 +21,13 @@ jobs:
         - python-version: "3.10"
           env:
             TOXENV: py
-        - python-version: "3.10"
```
Comment (wRAR) [inline .github/workflows/tests-windows.yml:24]: Not sure why we had asyncio for 3.10 and 3.11, if this was on purpose please chime in.

**Comment by wRAR** [comment]
I guess we need to wait until 3.12 is on the CI servers.

File: .github/workflows/tests-windows.yml (hunk 1 above)
Comment (Gallaecio) [reply .github/workflows/tests-windows.yml:24]: I had a look, and I believe it was by accident, https://github.com/scrapy/scrapy/commit/b6541830849a6edd41da6b539d272790a4661d7d#diff-ab5a8d3ff84c9430ab7fb358981634187b850ef3a5369ee94c332d59fd79dd93 followed by https://github.com/scrapy/scrapy/commit/8a0a9e6d3e5398b884e92cb9336efe3fd19677d6.

**Comment by Gallaecio** [comment]
The Windows issue may be legit, related to changes in URL parsing in 3.12 that are more strict. I think even for Windows paths, `file://` URLs need to use `/` as path separator, which may be causing the issue.

**Comment by wRAR** [comment]
Looks like the immediate cause is the presence of a colon in the path. Also, there is w3lib in the traceback, we should check its tests too.

**Comment by wRAR** [comment]
Though we don't run w3lib tests on Windows.

File: tests/test_downloader_handlers.py [hunk 2]
```diff
@@ -127,7 +127,7 @@ def _test(response):
         return self.download_request(request, Spider("foo")).addCallback(_test)
 
     def test_non_existent(self):
-        request = Request(f"file://{self.mktemp()}")
+        request = Request(path_to_file_uri(self.mktemp()))
```
Comment (wRAR) [inline tests/test_downloader_handlers.py:130]: The previous test is already doing this.

**Review by Gallaecio — APPROVED** [review]
:rocket:

**Pull request merged**